app/main/views/login.py

```diff
@@ -146,9 +182,178 @@ def update_password(token):
+
+
+@main.route('/update-user/<string:encoded_token>', methods=["POST"])
+def submit_update_user(encoded_token):
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    token = decode_invitation_token(encoded_token)
+    if token is None:
+        flash('token_invalid', 'error')
+        return render_template(
+            "auth/update-user.html",
+            token=None,
+            email_address=None,
+            supplier=None,
+            **template_data), 400
+    else:
+        user = data_api_client.get_user(email_address=token.get("email_address"))
+        data_api_client.update_user(
+            user_id=user["users"]["id"],
+            supplier_id=token.get("supplier_id"),
+            role='supplier'
+        )
+        user = User.from_json(user)
+        login_user(user)
+        return redirect(url_for('.dashboard'))
+
+
+@main.route('/create-user/<string:encoded_token>', methods=["POST"])
+def submit_create_user(encoded_token):
+    template_data = main.config['BASE_TEMPLATE_DATA']
+    form = CreateUserForm()
+
+    token = decode_invitation_token(encoded_token)
+    if token is None:
+        flash('token_invalid', 'error')
+        return render_template(
+            "auth/create-user.html",
+            form=form,
+            token=None,
+            email_address=None,
+            supplier=None,
+            **template_data), 400
+    else:
+        if form.validate_on_submit():
+            user = data_api_client.create_user({
+                'name': form.name.data,
+                'password': form.password.data,
+                'emailAddress': token.get('email_address'),
+                'role': 'supplier',
+                'supplierId': token.get('supplier_id')
+            })
+            user = User.from_json(user)
+            login_user(user)
+
+            return redirect(url_for('.dashboard'))
+        else:
+            return render_template(
+                "auth/create-user.html",
+                valid_token=False,
+                form=form,
+                token=encoded_token,
+                email_address=None,
+                supplier=None,
+                **template_data), 400
+
+
+@main.route('/invite-user', methods=["GET"])
+@login_required
+def invite_user():
+    form = EmailAddressForm()
+
+    template_data = main.config['BASE_TEMPLATE_DATA']
+
+    return render_template(
+        "auth/submit-email-address.html",
+        form=form,
+        **template_data), 200
+
+
+@main.route('/invite-user', methods=["POST"])
+@login_required
+def send_invite_user():
+    form = EmailAddressForm()
+
+    if form.validate_on_submit():
+
+        token = generate_token(
+            {
+                "supplier_id": current_user.supplier_id,
+                "supplier_name": current_user.supplier_name,
+                "email_address": form.email_address.data
+            },
+            current_app.config['SECRET_KEY'],
+            current_app.config['RESET_PASSWORD_SALT']
```

Comment:
Salts are used for token namespacing, so this should be different from the reset password one